src/client/actions.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Actions are the mechanism through which notifications send messages back to the application that raised them. The
+ * service defines a number of ways in which actions can be raised (typically user interactions, such as clicking a
+ * button), and it is up to each application to decide if it wishes to be informed of these interactions.
+ *
+ * For an application to receive an event on each of these actions, the application should specify the message it
+ * wishes to receive when that event happens. This is known as the action {@link NotificationActionResult|result}.
+ *
+ * All actions passed back to an application come through a single entry point. The application should attach a
+ * listener for the {@link NotificationActionEvent|`notification-action`} event. This listener will be called once for
+ * each action raised by the service.
+ *
+ * For an overview of actions, consider the sample notification below:
+ * ```ts
+ * import {addEventListener, create} from 'openfin-notifications';
+ *
+ * // Create a notification with two buttons
+ * create({
+ *     // Basic info
+ *     title: 'Reminder',
+ *     body: 'Event "Weekly Meeting" is starting soon...',
+ *     category: 'Upcoming Events',
+ *
+ *     // We'll use the 'customData' field to store metadata about the event
+ *     customData: {eventId: '12345'},
+ *
+ *     // We want the user clicking the notification to open the associated event, so register an 'onSelect' action
+ *     onSelect: {action: 'view-event', target: 'popup'},
```

Comment:
Specify what the significance of "view-event" and "popup" are?

Reply:
Totally arbitrary... The idea is this is application-specific data. There's another example that references these fields though - and in that example, these are a little more relevant. See doc comment on `NotificationActionResult`.